configs/components/yaml-cpp.rb

```diff
@@ -23,9 +23,14 @@
   elsif platform.is_macos?
     cmake_toolchain_file = ""
     cmake = "/usr/local/bin/cmake"
-    if platform.is_cross_compiled?
+    pkg.environment "PATH", "$(PATH):/opt/csw/bin"
+    if platform.architecture == 'arm64'
       pkg.environment 'CXX', 'clang++ -target arm64-apple-macos11' if platform.name =~ /osx-11/
       pkg.environment 'CXX', 'clang++ -target arm64-apple-macos12' if platform.name =~ /osx-12/
+      pkg.environment 'CXX', 'clang++ -target arm64-apple-macos13' if platform.name =~ /osx-13/
+      if(platform.architecture == "arm64" and platform.os_version.to_i >= 13)
```

Comment:
We prefer using `&&` over `and` as the latter are control flow operators and have different precedence that the usual boolean logic operators.

```suggestion
      if platform.architecture == "arm64" && platform.os_version.to_i >= 13
```